jvav/utils.py

```diff
@@ -130,12 +130,36 @@ def write_html(resp: requests.Response):
 
 class JavDbUtil(BaseUtil):
     BASE_URL = "https://javdb.com"
-    BASE_URL_SEARCH = "https://javdb.com/search?q="
+    BASE_URL_SEARCH = BASE_URL + "/search?q="
+    BASE_URL_VIDEO = BASE_URL + "/v/"
+    BASE_URL_ACTOR = BASE_URL + "/actors/"
 
-    def get_ids_from_page(self, url: str) -> typing.Tuple[int, list]:
-        """从某个页面获取番号列表
+    # 转换ID为JavDB ID
+    def get_javdb_id_by_id(self, id: str) -> typing.Tuple[int, None] | typing.Tuple[int, typing.Any]:
+        """
+        通过番号获取JavDB内部ID(实际是靠搜索实现的)
+        :param id: 番号
+        """
+        code, resp = self.send_req(url=JavDbUtil.BASE_URL_SEARCH + id)
+        if code != 200:
+            return code, None
+        try:
+            soup = self.get_soup(resp)
+            items = soup.find_all(class_="item")
+            for item in items:
+                if item.find(class_="video-title").strong.text.strip() == id.upper():
+                    return 200, item.find("a")["href"].split("/")[-1]
+            else:
```

Comment:
if else error here, please test again after you submit the code every time

Reply:
* This isn't an error, This is `for .... else` statement. The `else` clause is only executed when the loop statement has been executed without exiting midway.
* This code uses a `for` loop to check all search results, and only returns `(200, <javdb_id>)` when the ID of a certain result matches the input. Only execute the `else` (return 404) when all search results do not match. I think that's reasonable.
> More about this statement, you can refer : <https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops>

Reply:
Fine, however, this usage of python is not recommanded in _effective python_, and I think it is kind of counterintuitive , please implement in other way, no in for..else way